### PR TITLE
Adds 3D platformer tutorial.

### PR DESCRIPTION
**Pull request by saricden**
I have worked through the tutorial and made sure all the instructions work as expected.
Note that in some places the tutorial-code diverges slightly from the 3D platformer's repo code to facilitate for updates to the BuildKit tools, etc.

Let me know if you'd like to update the repo code so it matches, I'd be happy to do so!

Thanks everyone! :)

### CODE REVIEW

**Comment by saricden** [comment]
Note to test, you can pull locally then run the new `npm run docs` command from the repo's root to serve the docs at `http://localhost:3000`. Thanks!!

File: docs/buildkit/tutorial_3d_platformer.md [hunk 1]
```diff
@@ -0,0 +1,1337 @@
+# Tutorial: 3D Platformer
+
+In today's tutorial we're going to develop a 3D platformer game which can be played in your web browser. This game will leverage the Hathora `BuildKit` client and server libraries to facilitate multiplayer functionality.
```
Comment (hpx7) [inline docs/buildkit/tutorial_3d_platformer.md]:
```suggestion
In this tutorial we're going to develop a 3D platformer game which can be played in your web browser. This game will leverage the Hathora `BuildKit` client and server libraries to facilitate multiplayer functionality.
```

File: docs/buildkit/tutorial_3d_platformer.md [hunk 2]
```diff
@@ -0,0 +1,1337 @@
...
+
+Note that this tutorial is aimed at developers who may not have experience with Hathora, TypeScript, or even multiplayer game development in general. If you're a more advanced developer, feel free to skip sections so long as you feel confident in your understanding of what's going on.
+
+## Repo
+
+The source code of the 3D platformer can be located [here](https://github.com/hathora/3d-platformer). The repo's README file will guide you through running the game on your local machine, and there is an existing build of the game deployed to the web which can be sampled [here](https://3d-platformer.netlify.app/).
+
+Upon completion of this tutorial, your game will look like this:
+
+![A screenshot of the completed 3D platformer demo game, running fullscreen in the browser.](https://user-images.githubusercontent.com/7004280/213311928-18d0ed21-d309-4a30-8c7c-36cf0655d0f5.png)
```
Comment (hpx7) [inline docs/buildkit/tutorial_3d_platformer.md]: can this screenshot include another player to indicate that it's multiplayer?

File: docs/buildkit/tutorial_3d_platformer.md [hunk 3]
```diff
@@ -0,0 +1,1337 @@
...
+
+![A webpage client generated by npm init hathora-buildkit, indicating that our project was successfully bootstrapped.](https://user-images.githubusercontent.com/7004280/213313891-86709999-868c-4fc2-a0e7-f99d834797de.png)
+
+A moment or so after loading the page, the "Message from server:" box should flash green and display "Hello Hathora!"
+
+If you see this message, congrats! You've successfully established a client-server connection!
+
+## Common files
+
+It is a common practice when building games with Hathora BuildKit to create a `common` folder in the root of your project. This folder typically contains any and all files which are used within both the game's client and server code.
```
Comment (hpx7) [inline docs/buildkit/tutorial_3d_platformer.md]:
```suggestion
It is a common practice when building games with Hathora BuildKit to create a `common` folder in the root of your project. This folder typically contains any and all files which are shared bween both the game's client and server code.
```

File: docs/buildkit/tutorial_3d_platformer.md [hunk 4]
```diff
@@ -0,0 +1,1337 @@
...
+
+```ts
+// ...
+// Create an object to represent our Store
+const store: Store = {
+  // newState is called when a user requests a new room, this is a good place to handle any world initialization
+  newState(roomId: bigint, userId: string): void {
+    // const clock = new ServerClock();
+
+    // clock.onTick(delta => this.update(delta));
```
Comment (hpx7) [inline docs/buildkit/tutorial_3d_platformer.md]: should these be removed?

File: docs/buildkit/tutorial_3d_platformer.md [hunk 5]
```diff
@@ -0,0 +1,1337 @@
...
+    player.body.body.transform();
+
+    // Forward / backward movement
+    const {theta} = player;
+    const x = Math.sin(theta + (RADIANS_45 * player.direction.x)) * PLAYER_MOVE_SPEED * player.direction.z * delta;
+    // const x = PLAYER_MOVE_SPEED * player.direction.z * delta;
+    // const x = player.body.body.velocity.x;
+    const y = player.body.body.velocity.y;
+    const z = Math.cos(theta + (RADIANS_45 * player.direction.x)) * PLAYER_MOVE_SPEED * player.direction.z * delta;
+    // const z = PLAYER_MOVE_SPEED * player.direction.z * delta;
```
Comment (hpx7) [inline docs/buildkit/tutorial_3d_platformer.md]: remove commented out lines

File: docs/buildkit/tutorial_3d_platformer.md [hunk 6]
```diff
@@ -0,0 +1,1337 @@
...
+
+Note that the map's ground (grass-textured object shown in the screenshot previously) is *not* included in this array.
+
+## Server dependencies
+
+For the specific game we're building today, we will need a few additional dependencies for our server's code to work with. These dependencies are:
+
+- [enable3d](https://www.npmjs.com/package/enable3d): The 3D framework we're using (which can be run in a headless mode, aka without a visual representation)
+- [@enable3d/ammo-on-nodejs](https://www.npmjs.com/package/@enable3d/ammo-on-nodejs): The 3D physics engine utilized by enable3d
+- [hash.js](https://www.npmjs.com/package/hash.js): A collection of hash functions in pure JavaScript (but *with* TypeScript bindings)
```
Comment (hpx7) [inline docs/buildkit/tutorial_3d_platformer.md]: no longer used

File: docs/buildkit/tutorial_3d_platformer.md [hunk 7]
```diff
@@ -0,0 +1,1337 @@
...
+
+The first chunk of code we'll add will import everything required by our server, and specify configuration variables as constants. After finishing the rest of this tutorial, feel free to come back and mess around with these variables and observe the results!
+
+```ts
+// @ts-ignore
+import _ammo from '@enable3d/ammo-on-nodejs/ammo/ammo.js';
+import Enable3D from '@enable3d/ammo-on-nodejs';
+import { register, Store, UserId, RoomId } from "@hathora/server-sdk";
+import dotenv from "dotenv";
+import hash from "hash.js";
```
Comment (hpx7) [inline docs/buildkit/tutorial_3d_platformer.md]: no longer used

File: docs/buildkit/tutorial_3d_platformer.md [hunk 8]
```diff
@@ -0,0 +1,1337 @@
...
+## Client dependencies
+
+As with this game's server code, some additional dependencies are required by our game's client in order for our code to function.
+
+The clientside dependencies we will be consuming are:
+
+- [enable3d](https://www.npmjs.com/package/enable3d): The same 3D framework we're using on the server, but we will use it here for visuals.
+- [three](https://www.npmjs.com/package/three): A popular 3D JavaScript library which we will integrate with `enable3d`.
+- [interpolation-buffer](https://www.npmjs.com/package/interpolation-buffer): A utility for implementing linear interpolation into our game's client.
+- [hash.js](https://www.npmjs.com/package/hash.js): The same collection of hashing functions we installed on our server.
```
Comment (hpx7) [inline docs/buildkit/tutorial_3d_platformer.md]: is this used?

File: docs/buildkit/tutorial_3d_platformer.md (hunk 2 above)
Comment (saricden) [reply docs/buildkit/tutorial_3d_platformer.md]: Updated to include 2 additional players in the viewport, from a similar angle.

**Pull request merged**